src/runtime/audience-action-flow.js

```diff
@@ -340,6 +351,64 @@ export class AudienceActionFlow {
   }
 
   /**
+   * Populates localStorage with PPS configuration parameters based on
+   * SurveyDataTransferRequest.
+   * @param {SurveyDataTransferRequest} request
+   * @return {boolean}
+   * @private
+   **/
+
+  async configureAnswerPpsData(request) {
+    const iabAudienceKey = 'iabtaxonomiesvalues';
+    // PPS value field is optional and category may not be populated
+    // in accordance to IAB taxonomies.
+    const ppsConfigParams = request
+      .getSurveyQuestionsList()
+      .filter(
+        (question) => question.getSurveyAnswersList()[0].getPpsValue() !== null
+      )
+      .map((question) => question.getSurveyAnswersList()[0].getPpsValue());
+
+    if (ppsConfigParams.length > 0) {
+      const existingIabTaxonomy = await this.storage_.get(
+        iabAudienceKey,
+        /* useLocalStorage= */ true
+      );
+      let existingIabTaxonomyMap;
+      if (!existingIabTaxonomy) {
+        existingIabTaxonomyMap = {
+          '[googletag.enums.Taxonomy.IAB_AUDIENCE_1_1]': ppsConfigParams,
+        };
+      } else {
+        const existingIabMap = [
+          ...existingIabTaxonomy
+            .substring(existingIabTaxonomy.indexOf(':'))
+            .replace(/\D/g, ''),
+        ];
+        existingIabTaxonomyMap = {
+          '[googletag.enums.Taxonomy.IAB_AUDIENCE_1_1]':
+            existingIabMap.concat(ppsConfigParams),
+        };
+      }
+      await Promise.resolve(
+        this.storage_.set(
+          iabAudienceKey,
+          JSON.stringify(existingIabTaxonomyMap),
+          /* useLocalStorage= */ true
+        )
+      );
+      const result = Promise.resolve(this.storage_.get(iabAudienceKey, true));
+      return await !!Promise.resolve(
+        this.storage_.get(iabAudienceKey, /* useLocalStorage= */ true)
+      );
+    } else {
+      // Answers with no PPS parameters should always evaluate to true with
+      // the feature flag enabled.
+      return true;
```

Comment:
nit: can we move this null check to the beginning to simplify this control sequence?